src/docs/arch-cursor.dox

```diff
@@ -417,4 +417,33 @@ Thus, in the presence of a memory tracker, uses
 of "stray" pointers will be detected.  Even without a memory tracker, uses of "stray" pointers into
 the freed storage will likely yield the overwritten bytes, and not the previously seen key or value.
 
+@section arch_random_cursor Random Cursor
+
+WiredTiger uses a pseudo-random algorithm when using random cursors with the goal to achieve a good
+random distribution and being performant. Only the \c cursor->next function can be used to fetch a random record.
+The current pseudo-random algorithm is as follows:
+
+1. Starting from the root page we descend randomly down using WiredTiger's random number generator (RNG) until we
+reach a leaf page. Once we found a leaf page, we go into different conditions to grab a record from the page. 
+2. First condition: If the page has a sufficiently large number of disk records (1000 or more entries), attempt to find a 
+random visible disk record from the page using WiredTiger's RNG.
+3. Second condition: Loop through all insert lists, and find one that is large (1000 or more entries), once found attempt
+to find a random insert record from the page.
+4. Third condition: If the page has small amount of disk records (200 or less entries) or the leaf page is clean, attempt to
+find a random visible disk record from the page using WiredTiger's RNG.
+5. If all conditions fail, move onto the next or previous visbible record in the btree a random number of times
+between 0 and 250. There we can return the record that the cursor stopped on. Note: This can be on different leaf
+page than our original page that we descended from.
+
+The algorithm doesn't guarantee a complete randomness to the user and can lead to skewed results depending on the
+underlying tree structure. Usually cases when the tree structure is unbalanced, the records are not uniform, 
```

Comment:
Suggestion:

"Examples of instances that could typically lead to these skewed results are:

- An unbalanced btree.
- Records within a btree that aren't uniformly distributed.
- A high number of deleted items within a btree.
- A high number of invisible records in a transaction. "

Reply:
Done thank you!